yksgi/src/lib.rs

```diff
@@ -283,7 +285,8 @@ impl SGInterp {
     /// Implement call instructions. Returns `true` if the control point has been reached,
     /// `false` otherwise.
     unsafe fn call(&mut self) -> bool {
-        let func = LLVMGetCalledValue(self.pc.get());
+        let call_inst = self.pc.get();
+        let func = LLVMGetCalledValue(call_inst);
```

Comment:
Regarding my comment below I would probably change this to:
```
        let func = LLVMGetCalledValue(self.pc.get());
```
and leave `self.pc` as is, and replace `call_inst` with `self.pc/self.pc.get()` below.

Reply:
I tend to agree: the unusual name of the variable isn't helpful for readers.

Reply:
Should be addressed by d0327ba5d, albeit by a slightly different means.

Reply:
I like how you ignored what @ltratt and I both agreed on (replacing `call_inst` with `self.pc`, and instead did something completely different. :D I guess it's fine now, though I didn't really have an issue with using the `LLVM*` calls. I tried to only make wrappers for those that I thought will be used more than once. I didn't think we needed one for `LLVMIsFunctionVarArg` for example.

Anyway, let's see what @ltratt thinks.

Reply:
*ignored* is a bit strong. Truth be told, once I had added and used wrappers for various things, it made more sense to do it like this.

Reply:
Wait, having re-read the conversation, I'm confused.

> unusual name of the variable 

What is unusual about `call_inst`? `self.pc` is a pointer to a `CallInst` instruction.

(I introduced the variable to avoid repeatedly doing `self.pc` everywhere)

Reply:
It's not too unusual for me, but I still think it's unneccesary (we don't do this anywhere else in the code either).

You added `let call_inst = self.pc` so you don't have to repeatedly type `self.pc` but instead have to repeatedly type `const_inst` which is even 3 characters longer? ;-)

Reply:
It's not about making it short. It's about avoiding (potentially) extra field lookups, and it also makes clear that `self.pc` doesn't change in the interim.

Reply:
Fair enough. Though the compiler is going to remove the field lookup anyway. It's just that the rest of the code never does this, so I just wanted to keep it in style. If you feel strongly about it I'm happy for you to leave it, if @ltratt is happy too.

Reply:
Let me put it differently.

If I gave you code like:
```
f(a + b)
g(a + b)
h(a + b)
```

Wouldn't you change it to:
``
let sum = a + b;
f(sum)
g(sum)
h(sum)
```

?

Of course you would :P Even if the compiler can do common sub-expression elimination, it just makes for nicer code.

Reply:
> Though the compiler is going to remove the field lookup anyway.

We raced, but I preempted this comment :P

Reply:
I would argue that `self.pc` isn't the same as `a + b`. But I think this isn't worth to argue about, so I'll leave the decision to @ltratt.

Reply:
Of course @vext01 is right in general, but in this instance, I *really* want to know if you're using the pc or not. IMHO either call the variable `pc` or use `self.pc` (the two are probably optimised to the same, so it's a matter of style).

Reply:
Is the confusion stemming from the fact that `self.pc` is a program counter in the sense that it is a pointer to the current instruction?

When we are in the `call()` method of the interpreter we know that `self.pc` can only point to an instruction that does some kind of a call, hence `self.pc` is a pointer to an instruction that makes some kind of call, hence `call_inst`.

Reply:
I understand that reasoning in one way -- but what's the point of that degree of indirection, when it confuses the code reader, and there's a clearer equivalent which requires no more characters to type?

Reply:
I find it much clearer to narrow the general `self.pc` into a more descriptive name, given the invariant of the  context.

In fact, I find the name `self.pc` confusing itself. I think it should be `self.cur_inst`.

Reply:
If you want to change `pc` to `cur_inst` that's a debate / PR we can have -- but this isn't the right place to do it. Consistency has a virtue in and of itself, and `cur_inst` just doesn't sit with the rest of the code base. Sorry to be a dick about it!

Reply:
It's not a hill I'm willing to die on, so I'll back down and quietly fume in a corner :P

Reply:
You have my permission to print my ugly mug onto a dartboard :)

Reply:
Fixed in fc190961fe2d